### PR TITLE
user can add and view promos with location data included

### PR DESCRIPTION
**Pull request by Cody-Hayes97**
# Description
updated user promotions so user can add promos rendered with location info
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ x] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [x ] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [ ] Test A
- [ ] Test B

# Checklist

- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x ] New and existing unit tests pass locally with my changes
- [ x] There are no merge conflicts

### CODE REVIEW

File: src/components/users/promotions/UserPromos.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+} from "../../../redux/actions/userActions";
+
+export default function UserPromos() {
+  const user = useSelector(({ user }) => user);
+  const savedPromos = useSelector(({ user }) => user.savedPromos);
+  const location = useSelector((state) => state.location);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    console.log(savedPromos, "from dispatch");
```
Comment (agyin3) [inline src/components/users/promotions/UserPromos.js:17]: do you think we can remove this console.log?

File: src/redux/actions/userActions.js [hunk 2]
```diff
@@ -237,6 +237,34 @@ export const deleteUserFriends = (id, user) => (dispatch) => {
...
+        payload: res.data,
+      });
+    })
+    .catch((err) => {
+      console.log(err);
+    });
+};
+
+export const addUserPromo = (user_id, promo_id) => (dispatch) => {
+  // dispatch({ type: types.GET_USER_FRIENDS_START, payload: true });
```
Comment (agyin3) [inline src/redux/actions/userActions.js:257]: do you think we can remove this unused code?

**Comment by agyin3** [comment]
Great job adding this functionality 👍

File: src/redux/actions/userActions.js (hunk 2 above)
Comment (Cody-Hayes97) [reply src/redux/actions/userActions.js:257]: yeah, thanks for catching both of those

**Pull request merged**